modules/platform/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/rendering/operations/RenderDocument.java

```diff
@@ -65,7 +65,8 @@
 
     @OperationMethod(collector = BlobCollector.class)
     public Blob run(DocumentModel doc) throws OperationException, RenderingException, TemplateException, IOException {
-        String content = RenderingService.getInstance().render(type, template, ctx);
+        OperationContext subCtx = ctx.getSubContext(true, doc);
```

Comment:
I don't know this code well but all the uses of `getSubContext` I can find all use try-with-resources to close it when done.